yb-voyager/cmd/callhome_payload_builder.go

```diff
@@ -0,0 +1,219 @@
+		PayloadVersion:                 callhome.ASSESS_MIGRATION_CALLHOME_PAYLOAD_VERSION,
+		TargetDBVersion:                assessmentReport.TargetDBVersion,
+		Sizing:                         &callhomeSizingAssessment,
+		MigrationComplexity:            assessmentReport.MigrationComplexity,
+		MigrationComplexityExplanation: assessmentReport.MigrationComplexityExplanation,
+		SchemaSummary:                  callhome.MarshalledJsonString(schemaSummaryCopy),
+		Issues:                         anonymizedIssues,
+		Error:                          callhome.SanitizeErrorMsg(errMsg),
+		TableSizingStats:               callhome.MarshalledJsonString(tableSizingStats),
+		IndexSizingStats:               callhome.MarshalledJsonString(indexSizingStats),
+		SourceConnectivity:             assessmentMetadataDirFlag == "",
+		IopsInterval:                   intervalForCapturingIOPS,
+		ControlPlaneType:               getControlPlaneType(),
+		AnonymizedDDLs:                 getAnonymizedDDLs(&source),
+	}
+
+	payload.PhasePayload = callhome.MarshalledJsonString(assessPayload)
+	err = callhome.SendPayload(&payload)
+	if err == nil && (status == COMPLETE || status == ERROR) {
+		callHomeErrorOrCompletePayloadSent = true
+	}
+}
+
+func anonymizeAssessmentIssuesForCallhomePayload(assessmentIssues []AssessmentIssue) []callhome.AssessmentIssueCallhome {
+	/*
+		Case to skip for sql statement anonymization:
+			1. if issue type is unsupported query construct or unsupported plpgsql object
+			2. if object type is view or materialized view
+			3. if sql statement is empty
+	*/
+	shouldSkipAnonymization := func(issue AssessmentIssue) bool {
+		skipCategories := []string{UNSUPPORTED_QUERY_CONSTRUCTS_CATEGORY, UNSUPPORTED_PLPGSQL_OBJECTS_CATEGORY, UNSUPPORTED_DATATYPES_CATEGORY}
+		skipObjects := []string{constants.VIEW, constants.MATERIALIZED_VIEW, constants.TRIGGER, constants.FUNCTION, constants.PROCEDURE}
+
+		return slices.Contains(skipCategories, issue.Category) ||
+			slices.Contains(skipObjects, issue.ObjectType) ||
+			issue.SqlStatement == ""
+	}
+
+	anonymizedIssues := make([]callhome.AssessmentIssueCallhome, len(assessmentIssues))
+	for i, issue := range assessmentIssues {
+		anonymizedIssues[i] = callhome.NewAssessmentIssueCallhome(issue.Category, issue.CategoryDescription, issue.Type, issue.Name, issue.Impact, issue.ObjectType, issue.Details)
+
+		// special handling for extensions issue: adding extname to issue.Name
+		if issue.Type == queryissue.UNSUPPORTED_EXTENSION {
+			anonymizedIssues[i].Name = queryissue.AppendObjectNameToIssueName(issue.Name, issue.ObjectName)
+		}
+
+		if shouldSkipAnonymization(issue) {
+			continue
+		}
+
+		var err error
+		anonymizedIssues[i].SqlStatement, err = anonymizer.AnonymizeSql(issue.SqlStatement)
+		if err != nil {
+			log.Warnf("failed to anonymize sql statement for issue %s: %v", issue.Name, err)
+		}
+	}
+
+	return anonymizedIssues
+}
+
+func getAnonymizedDDLs(sourceDBConf *srcdb.Source) []string {
```

Comment:
Now, when I checked this again, isn't this `sourceDBConf` is not required to be passed in the parameter, as source is the global variable and it is accessible here as well directly?